File: Mechanix.Test/ForceTest.cs
using System.Runtime.Serialization.Formatters.Binary;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using static Microsoft.VisualStudio.TestTools.UnitTesting.Assert;

namespace Mechanix.Test
{
    [TestClass]
    public class ForceTest
    {
        [TestMethod]
        public void TestOperators()
        {
            Force f1 = new Force(1, 3, -2);
            Force f2 = new Force(0, -1, 1);

            AreEqual(new Force(1, 2, -1), f1 + f2);
            AreEqual(new Force(1, 2, -1), f1.Add(f2));
            AreEqual(new Force(1, 2, -1), f2.Add(f1));

            AreEqual(new Force(1, 4, -3), f1 - f2);
            AreEqual(new Force(1, 4, -3), f1.Substract(f2));
            AreEqual(new Force(1, 4, -3), Force.Zero - f2.Substract(f1));

            AreEqual(Force.Zero, f1.Multiply(0));
            AreEqual(Force.Zero, f1 * 0);
            AreEqual(new Force(2, 6, -4), f1 * 2);
        }

        [TestMethod]
        public void TestSerialization()
        {
            var f = new Force(12, 222, 0.2);

            var formatter = new BinaryFormatter();
            var stream = new System.IO.MemoryStream();

            formatter.Serialize(stream, f);
            stream.Position = 0;
            var newf = formatter.Deserialize(stream);

            AreEqual(f, newf);
        }
    }
}

File: Mechanix.PropertyTest/ForceTest.fs
namespace Mechanix.PropertyTest

open FsCheck
open FsCheck.Xunit
open Mechanix
open System.Runtime.Serialization.Formatters.Binary

type ForceGenerator =
    static member Gen() =
        Arb.generate<double>
        |> Gen.three
        |> Gen.map (fun (x, y, z) -> Force(x, y, z))
        |> Arb.fromGen

type ForceTest() =
    do Arb.register<ForceGenerator>() |> ignore

    [<Property>]
    let ``Addition is transitive`` (f1 : Force) (f2 : Force) =
        f1.Add(f2).Equals(f2.Add(f1))

    [<Property>]
    let ``Deserialized equals origin`` (f : Force) =
        let formater = new BinaryFormatter()
        let stream = new System.IO.MemoryStream()
        do formater.Serialize(stream, f)
        do stream.Position <- int64 0
        f.Equals(formater.Deserialize stream)
        